### ISSUE TITLE
Enhanced Open Enterprise template

### ISSUE THREAD

**Issue by javieralaves** [AutarkLabs/product#3]
## Description

Releasing a new template or enhancing Open Enterprise such that it permits an additional Tokens instance, allowing either of the two Tokens instances to be configured as defined by the organization.

### Acceptance criteria

- [x] An additional Tokens app instance can be initiated
- [x] Either of the two Tokens app instances can be configured flexibly
- [x] Either of the two Tokens apps can be referenced to govern over the Voting or Dot Voting app
- [x] Agent app is available as an optional app for install

What's not to be included in this epic:
* Any new Help components, which need to be further defined in a different epic

## Product specs

[Available on Google Drive](https://docs.google.com/document/d/1fD3MBO7cLfnig1r9OtlMDx3IpsXsO2GyavCOPIEvwNs/edit?usp=sharing).

**Comment by javieralaves** [comment]
@stellarmagnet Ready for your review. This latest iteration covers perhaps the simplest solution to the two-Tokens template design, which would be to add a Help icon for contextual information. I wasn't able to find the right fit for something different, for now. I've also defined the Tokens initialization parameters, similar to what you defined a few weeks ago.

**Comment by stellarmagnet** [comment]
Hello Javi! I added some comments to the Figma. One idea is we can simply bring back one of the previous designs you had, but just reword it so the user is choosing between 1 or 2 token org. I'm just not sure if the current way with the additional "Switch" for Tokens would be intuitive...

I think the switch does make sense for adding the Agent. I feel we should add a help icon there too, as first time users may not know the benefits of adding the agent.

I added a few comments to the spec, some things we need to clarify with development. I think we have to allow the user to select which token is tied to Voting or Dot Voting. We are going to be assuming here that only one voting app is needed (e.g. they are only mapping it to one token) although we are allowing mixed governance between Voting and Dot Voting.

**Comment by javieralaves** [comment]
@stellarmagnet This is ready for product review!

**Comment by Quazia** [comment]
@ottodevs is going to review Peters work, and handle the client implementation to the spec. For the contract @PeterMPhillips can start by moving the testing from Otto's branch over to his template implementation and add any additional tests needed. From there we just need to confirm that the contract work is in sync with the needs from the Gdocs specification.

**Comment by stellarmagnet** [comment]
@javieralaves looks good to me, there are just a few design comments in the Figma that I think shouldn't take that long to resolve. Moving this to technical review.

**Comment by javieralaves** [comment]
@stellarmagnet Figma and specs updated, this is ready to go!

**Comment by ottodevs** [comment]
Hi @javieralaves ! While implementing the UI for this, closely following your figma design, I detected an unusual size for the 1 token asset, this one:
![image](https://user-images.githubusercontent.com/5030059/70341803-0b0de180-1854-11ea-92ad-59dab406c232.png)

Just removing the shirt pattern of the sitting person in the illustration, the size goes from 2.2MB to 8KB, so I was wondering if we might want to optimize that asset.

I also detected unusual sizes in other open-enterprise assets using the same pattern, specifically these ones:
![image](https://user-images.githubusercontent.com/5030059/70341892-3a245300-1854-11ea-9cf3-d6f2fb82baa1.png)

Basically the empty state card illustrations for those 3 apps.

It is probably barely noticeable with the modern fast connections, but I think we still could try to optimize that, even for processing those files from the code perspective it may result slow to manage and we seem to be defeating the vector graphics advantages, as these images are actually heavy `png` files embedded into an `svg` `pattern`

If you find it is not evident how to improve the assets, I can try looking into any cli or online tool for image optimization, just ping me with your thoughts 👍

**Comment by javieralaves** [comment]
Hey @ottodevs, this is definitely of relevance, thank you for point this out! I'll keep a better eye on this aspect in the future and send you optimized assets today or tomorrow.

**Comment by ottodevs** [comment]
> Hey @ottodevs, this is definitely of relevance, thank you for point this out! I'll keep a better eye on this aspect in the future and send you optimized assets today or tomorrow.

FYI during last weekend I was playing a bit with these assets and I manage to reduce their size by 10 times by following this methodology:
- Copy the embedded `png` pattern (encoded into `base64`) and use online tool to decode the actual `png` image 
- Use another online tool to resize and optimize `png` images, reducing the size over 90%
- Encode again into a `base64` string
- Paste again in the `svg` markup `pattern` layer

Of course if we do this during design phase, those steps would be probably unnecessary, but maybe if you cannot figure out a better way, it might do the trick for now.

Also looking on the A1 take on this, I realized they just use strongly optimized `png` files for the aragon client instead of using `svg` at all... but I guess -and this is a personal opinion- that `svg` is a better format for these kind of assets, because the scalability for high density resolutions.

**Comment by javieralaves** [comment]
Thanks @ottodevs! And sorry that I was unable to get these to you earlier. I agree with your opinion about using `svg` over `png` as a file format. I'll make sure that either way future graphic assets are as optimized for deployment as they can be.

**Comment by ottodevs** [comment]
Pending review, it is blocked by #20 and AutarkLabs/open-enterprise#1532 but the contracts/UI regarding:
- AutarkLabs/open-enterprise#1524
- AutarkLabs/open-enterprise#1604
can definitely be reviewed at AutarkLabs/open-enterprise#1597 and:
- AutarkLabs/open-enterprise#1367
- AutarkLabs/open-enterprise#1533
can be reviewed at https://beta.autark.xyz and https://github.com/aragon/aragon/pull/1233
